scripts/pre-commit.sh

```diff
@@ -22,23 +22,32 @@ set -o pipefail
 
 source "$(dirname "${BASH_SOURCE}")/util.sh"
 ROOT_DIR="$(cd "$(dirname "$0")/.." ; pwd)"
+TEMP_DIR="$(mktemp -d)"
 MAKE_CMD="make -C ${ROOT_DIR}"
+OS="$(go env GOOS)"
+ARCH="$(go env GOARCH)"
+PLATFORM="${OS}-${ARCH}"
 NUM_CLUSTERS="${NUM_CLUSTERS:-2}"
 JOIN_CLUSTERS="${JOIN_CLUSTERS:-}"
 DOWNLOAD_BINARIES="${DOWNLOAD_BINARIES:-}"
 CONTAINER_REGISTRY_HOST="${CONTAINER_REGISTRY_HOST:-172.17.0.1:5000}"
 COMMON_TEST_CMD="go test -v"
-COMMON_TEST_ARGS="./test/e2e -args  -kubeconfig=${HOME}/.kube/config -ginkgo.v -single-call-timeout=1m -ginkgo.trace -ginkgo.randomizeAllSpecs"
-E2E_TEST_CMD="${COMMON_TEST_CMD} ${COMMON_TEST_ARGS}"
+COMMON_BINARY_TEST_CMD="${TEMP_DIR}/e2e-${PLATFORM}"
+COMMON_TEST_ARGS="-kubeconfig=${HOME}/.kube/config -ginkgo.v -single-call-timeout=1m -ginkgo.trace -ginkgo.randomizeAllSpecs"
+E2E_BINARY_TEST_CMD="${COMMON_BINARY_TEST_CMD} ${COMMON_TEST_ARGS}"
```

Comment:
(No action required) Maybe rename to `E2E_TEST_CMD` since the in-memory command is already differentiated?